contracd/src/providediagnostickeys.cpp

```diff
@@ -253,27 +256,35 @@ void ProvideDiagnosticKeys::run()
         }
         terminate = shouldTerminate();
     }
+    qDebug() << "building keys list took " << buildKeysTimer.elapsed() << "ms";
 
     for (qint32 day = 0; (day < DAYS_TO_STORE) && !terminate; ++day) {
         qDebug() << "Aggregating day: " << day;
         if (diagnosisKeys[day].length() > 0) {
             quint32 const dayNumber = static_cast<quint32>(m_currentDayNumber + day);
             qDebug() << "Calling findDtkMatches with" << diagnosisKeys[day].count() << " keys";
 
+            QElapsedTimer findingMatchesTimer;
+            findingMatchesTimer.start();
             QList<ContactMatch> matches;
             qint32 batchPos = 0;
             while (batchPos < diagnosisKeys[day].size() && !terminate) {
                 QList<DiagnosisKey> diagnosticKeyBatch;
                 diagnosticKeyBatch = diagnosisKeys[day].mid(batchPos, MAX_BATCH_SIZE);
                 batchPos += MAX_BATCH_SIZE;
-                matches.append(m_d->m_contacts->findDtkMatches(dayNumber, diagnosisKeys[day]));
+                matches.append(m_d->m_contacts->findDtkMatches(dayNumber, diagnosticKeyBatch));
```

Comment:
this line contained a bug which lead to calling findDtkMatches with way too large lists and also calling it a bunch of times with the same data...

Reply:
i did some quick calculations,

currently i have #diagnosiskeys:
17773 keys -> 4983ms
36637 keys -> 14406ms
...
68117 keys -> 34108ms
69253 keys -> 34716ms
72814 keys -> 35664ms

i did not measure all the old durations, but assuming that every call of findDtkMatches would take above durations (since the input would be **all** keys), i would get
17773 keys -> 18 calls * 4983ms --> 89694ms
36637 keys -> 36 calls * 14406ms --> 518616ms (~8.6 minutes)
...
68117 keys -> 67 calls * 34108ms --> 2285236ms (~38 minutes)
69253 keys -> 68 calls * 34716ms --> 2360688ms (~39 minutes)
72814 keys -> 72 calls * 35664ms --> 2567808ms (~43 minutes)
remember, this is node for 14 days (so improvement is like 14 * 0.5 = 7 minutes instead of 14 * 30 = 420 Minutes => (~7hours)

taking the real measurements, i find that this matches the measured times quite well...

real measurement for a few "old" calls (the first ones since the others are taking waaaay too loong)
17773 keys --> 90756ms
36637 keys --> 519933ms
...